src/features/imageUpload/UploadButton.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { connect } from "react-redux";
+import styled from "styled-components";
+import { getActiveRoomId } from "../../selectors/selectors";
+import { appendText } from "../input/chatInputReducer";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import imageLoader from "./imageLoader.js";
+import imageUpload from "./imageUpload.js";
+
+const UploadContainer = styled.a`
+	display: inline-block;
+	background-color: transparent;
+	position: relative;
+	cursor: pointer;
+`;
+
+const FileUpload = styled.input`
+	position: absolute;
+	top: 0;
+	left: 0;
+	width: 100%;
+	height: 100%;
+	opacity: 0;
+`;
+
+const mapStateToProps = state => ({
+	activeRoomId: getActiveRoomId(state)
+});
+
+const mapDispatchToProps = {
+	appendText
+};
+
+class UploadButton extends React.PureComponent {
+
+	fileUploadElement = null;
+
+	constructor(props) {
+		super(props);
+		this.state = { uploading: false };
```

Comment:
state be declared as 
`
state = { uploading: false }
`
(dont need a constructor)

Reply:
This would be a same as `fileUploadElement = null` above